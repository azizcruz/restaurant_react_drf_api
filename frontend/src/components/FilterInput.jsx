import React, { Component } from "react";
import { Input } from "antd";
import { Row, Col } from "antd";

const Search = Input.Search;

export class FilterInput extends Component {
  render() {
    return (
      <div>
        <Row>
          <Col sm={{ span: 12 }} md={{ span: 6 }} lg={{ span: 4 }}>
            <Search
              placeholder="Filter by name"
              onSearch={value => this.props.onFilter(value)}
              enterButton
              style={{ marginBottom: 24 + "px" }}
            />
          </Col>
        </Row>
      </div>
    );
  }
}

export default FilterInput;
