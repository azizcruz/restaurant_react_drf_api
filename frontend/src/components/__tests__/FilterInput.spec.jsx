import React from "react";
import { create } from "react-test-renderer";
import FilterInput from "../FilterInput";

describe("FilterInput component", () => {
    test("it matches the snapshot", () => {
      const component = create(<FilterInput />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });