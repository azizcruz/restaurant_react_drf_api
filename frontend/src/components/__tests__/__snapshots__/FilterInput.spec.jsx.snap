// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLayout component it matches the snapshot 1`] = `
<div>
  <div
    className="ant-row"
  >
    <div
      className="ant-col ant-col-sm-12 ant-col-md-6 ant-col-lg-4"
    >
      <span
        className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Filter by name"
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-group-addon"
          >
            <button
              className="ant-btn ant-input-search-button ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                aria-label="icon: search"
                className="anticon anticon-search"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>
  <br />
  <br />
</div>
`;

exports[`DataTable component it matches the snapshot 1`] = `
<div>
  <div
    className="ant-row"
  >
    <div
      className="ant-col ant-col-sm-12 ant-col-md-6 ant-col-lg-4"
    >
      <span
        className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Filter by name"
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-group-addon"
          >
            <button
              className="ant-btn ant-input-search-button ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                aria-label="icon: search"
                className="anticon anticon-search"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>
  <br />
  <br />
</div>
`;

exports[`FilterInput component it matches the snapshot 1`] = `
<div>
  <div
    className="ant-row"
  >
    <div
      className="ant-col ant-col-sm-12 ant-col-md-6 ant-col-lg-4"
    >
      <span
        className="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Filter by name"
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-group-addon"
          >
            <button
              className="ant-btn ant-input-search-button ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                aria-label="icon: search"
                className="anticon anticon-search"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>
  <br />
  <br />
</div>
`;
